narwhal/worker/src/batch_maker.rs

```diff
@@ -115,10 +115,13 @@ impl BatchMaker {
                 // Note that transactions are only consumed when the number of batches
                 // 'in-flight' are below a certain number (MAX_PARALLEL_BATCH). This
                 // condition will be met eventually if the store and network are functioning.
-                Some((transaction, response_sender)) = self.rx_batch_maker.recv(), if batch_pipeline.len() < MAX_PARALLEL_BATCH => {
+                Some((transactions, response_sender)) = self.rx_batch_maker.recv(), if batch_pipeline.len() < MAX_PARALLEL_BATCH => {
                     let _scope = monitored_scope("BatchMaker::recv");
-                    current_batch_size += transaction.len();
-                    current_batch.transactions_mut().push(transaction);
+                    for transaction in transactions {
```

Comment:
Is it possible to respect the batch size limit and break the soft bundle? Otherwise batches can get really large. I believe most of the time bundles will not get broken up.

Alternatively, if the bundle size is less than batch size, we can create a batch for the bundle.